x/liquidity/amm/pool.go

```diff
@@ -124,11 +136,19 @@ func (pool *BasicPool) LowestSellPrice() (price sdk.Dec, found bool) {
 
 // BuyAmountOver returns the amount of buy orders for price greater than
 // or equal to given price.
-func (pool *BasicPool) BuyAmountOver(price sdk.Dec) sdk.Int {
+func (pool *BasicPool) BuyAmountOver(price sdk.Dec) (amt sdk.Int) {
 	if price.GTE(pool.Price()) {
 		return sdk.ZeroInt()
 	}
-	return pool.rx.ToDec().QuoTruncate(price).Sub(pool.ry.ToDec()).TruncateInt()
+	utils.SafeMath(func() {
+		amt = pool.rx.ToDec().QuoTruncate(price).Sub(pool.ry.ToDec()).TruncateInt()
+		if amt.GT(MaxCoinAmount) {
+			amt = MaxCoinAmount
+		}
+	}, func() {
+		amt = MaxCoinAmount
```

Comment:
Don't we need to add this logic on spec? If someone reach this logic without knowing this fact, someone will likely suffer losses

Reply:
Can you explain more about what *losses* means here?